### ISSUE TITLE
Online Co-op: ERROR: Creature.instantiatedTemporaryItem needs to be null first before calling EquipTemporaryItem – Part 2

### ISSUE THREAD

**Issue by ChrisNZL** [ChrisNZL/Tallowmere2#724]
Similar to issue #668, but this time, only being reported in 0.2.1, and seems to happen in the middle of a room, rather than being tied to any specific event or game state.

Feedback IDs:
- 20201211-FHH5V
- 20201211-WSGRZ
- 20201210-DHF2W

```
8:20:04, Frame 43679, LOG »  DungeonRoomTransitioner.Update: Revealing room: 第2个房间：下水道


8:20:39, Frame 48658, ERROR »  ERROR: Creature.instantiatedTemporaryItem needs to be null first before calling EquipTemporaryItem
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>


GameStates: RompingThroughDungeon
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 2 / 2
RoomModifiers: None
YouHaveDiedState: Null
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
In 0.2.2, added further debugging checks – extended the `MethodCaller` enum.

Will investigate further as future reports come in.

**Comment by ChrisNZL** [comment]
New reports in 0.2.2.

`MethodCaller` seems to be `NetworkMessage_PickUpTemporaryItem`.

Likely related: [Online Co-op: Home room key issue / pre-run syncing #770](https://github.com/ChrisNZL/Tallowmere2/issues/770)

Feedback IDs:
- 20201228-52NNY
- 20201227-Q9WQV

**To investigate:**
- `CreatureStruct.currentTemporaryItemPrefab` – probably is not getting set or reset somewhere.

---

```
12:02:54, Frame 2067, LOG »  DungeonRoomTransitioner.Update: Revealing room: Tallowmere的城堡


12:02:55, Frame 2112, ERROR »  ERROR: Creature.HoldTemporaryItemIfNeeded:
instantiatedTemporaryItem needs to be null first before calling HoldTemporaryItemIfNeeded.
methodCaller: NetworkMessage_PickUpTemporaryItem
>>>>> CRITICAL ERROR >>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>


GameStates: RompingThroughDungeon
OnlineGameInfo: Device is online.
Server: Lightning-Beijing-01
GameID: H627V
GameSetupMode: NetworkCoop
Players: 2
DungeonRoomTransitionerState: Null
Room: 0 / 0
RoomModifiers: None
YouHaveDiedState: Null
SystemPlayer InputDevice: 键盘
HumanPlayer 1 InputDevice: 键盘
```

**Comment by ChrisNZL** [comment]
Believe this is fixed now in 0.2.3, or at least less error-ish.

Restructured order of events within `NetworkMessage_PickUpTemporaryItem`, `NetworkMessage_ThrowTemporaryItem`, and `Creature.HoldTemporaryItemIfNeeded` to ensure things don't exit too early within certain methods. 

For the case of `MethodCaller.NetworkMessage_PickUpTemporaryItem`, changed this from an Error to a Warning, as I'm not sure it needs to error and cancel early, if it's just a network message mishap. Should let things continue on now when playing online. May be an ongoing investigation if key-holding issues still occur, but I haven't been able to reproduce this now.
